src/flinkStatementResultsManager.ts

```diff
@@ -222,19 +227,56 @@ export class FlinkStatementResultsManager {
     }
   }
 
+  /**
+   * Filters results based on the current search query and visible columns.
+   * If visibleColumns is undefined, searches through all columns.
+   * If visibleColumns is defined, only searches through the specified columns.
+   *
+   * @param results - Array of result rows to filter
+   * @param visibleColumns - Optional array of column names to search through. If undefined, searches all columns.
+   * @returns Filtered array of results that match the search query in the specified columns
+   */
+  private filterResultsBySearch(results: any[], visibleColumns: string[] | undefined): any[] {
+    const searchQuery = this._searchQuery();
+    if (searchQuery === null) {
+      return results;
+    }
+
+    const searchLower = searchQuery.toLowerCase();
+    return results.filter((row) =>
+      Object.entries(row)
+        .filter(([key]) => visibleColumns === undefined || visibleColumns.includes(key))
+        .some(([_, value]) => value !== null && String(value).toLowerCase().includes(searchLower)),
+    );
+  }
+
   handleMessage(type: MessageType, body: any): any {
     switch (type) {
       case "GetResults": {
         const offset = body.page * body.pageSize;
         const limit = body.pageSize;
         const paginatedResults = this.getResultsArray().slice(offset, offset + limit);
```

Comment:
Shouldn't pagination happen _after_ search filtering, not before?

If there are 10,000 unfiltered results, then I search filter down to only 100, pagination should be based on 100, not 10,000.

This implementation looks to leave me in sutuations showing only 5 rows when on page 50 of 100 pages, when should want to be showing 'the next 100 matching rows'.